text/0082-combined-replay-envelope-item.md

```diff
@@ -0,0 +1,100 @@
+- Start Date: 2023-03-24
+- RFC Type: informational
+- RFC PR: https://github.com/getsentry/rfcs/pull/82
+- RFC Status: draft
+
+# Summary
+
+Right now Session Replays data is sent from the SDK via two envelope item types:
+
+ReplayEvent and ReplayRecording.
+
+- ReplayEvent contains indexable metadata that is stored in our clickhouse table.
+- ReplayRecording contains a large, usually compressed blob that is written to our Filestore (GCS as of now)
+
+These are always sent in the same envelope by the sdk. We'd like to combine these two envelope item types into one.
+
+We'd first write a compatibility layer in relay that takes the old format and converts them into a single item type.
+We'd then modify the sentry-sdk to send combined item types going forward
```

Comment:
We need to consider compatibility with Self Hosted (and Single Tenant) before we change the SDK